src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/LanguageServer/RazorDynamicDocumentSyncRegistration.cs

```diff
@@ -66,7 +66,11 @@ public async Task OnInitializedAsync(ClientCapabilities clientCapabilities, Requ
                             {
                                 Id = Guid.NewGuid().ToString(), // No need to save this for unregistering
                                 Method = Methods.TextDocumentDidChangeName,
-                                RegisterOptions = registrationOptions
+                                RegisterOptions = new TextDocumentChangeRegistrationOptions()
+                                {
+                                    DocumentSelector = documentFilters,
+                                    SyncKind = TextDocumentSyncKind.Incremental
```

Comment:
The sync kind was more important here, but I went ahead and updated the documentFilter as well